File: HealthLink_BE/src/main/java/com/HealthLink/dto/pharmacy/PharmacyOrderRequest.java
package com.HealthLink.dto.pharmacy;

import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class PharmacyOrderRequest {

    /** Mã đơn thuốc (bắt buộc) */
    @NotNull(message = "Prescription header ID is required")
    private Integer prescriptionHeaderId;

    /** Mã nhà thuốc được chọn (bắt buộc) */
    @NotNull(message = "Pharmacy ID is required")
    private String pharmacyId;

    /**
     * Loại giao hàng: "Delivery" hoặc "Pickup" (mặc định: Delivery)
     */
    private String deliveryType = "Delivery";

    /** Địa chỉ giao hàng tùy chỉnh (nếu null, lấy từ Patient) */
    private String deliveryAddress;

    /** Vĩ độ giao hàng tùy chỉnh */
    private Double deliveryLatitude;

    /** Kinh độ giao hàng tùy chỉnh */
    private Double deliveryLongitude;

    /** Phương thức thanh toán */
    private String paymentMethod;

    /** Ghi chú của bệnh nhân/bác sĩ */
    private String notes;
}

File: HealthLink_FE/src/components/patient-dashboard/pharmacy-order-request.js
export function buildPrescriptionPharmacyRequestPayload({
  patientId,
  pharmacyId,
  prescriptionHeaderId,
  fulfillmentType,
  deliveryContact,
}) {
  if (!prescriptionHeaderId) {
    throw new Error('Prescription is required for a pharmacy order request.');
  }

  return {
    patientId,
    pharmacyId,
    requestType: 'ORDER_REQUEST',
    preferredDeliveryType: fulfillmentType,
    deliveryType: fulfillmentType,
    deliveryAddress: deliveryContact?.deliveryAddress,
    deliveryLatitude: deliveryContact?.deliveryLatitude,
    deliveryLongitude: deliveryContact?.deliveryLongitude,
    deliveryPhoneNumber: deliveryContact?.deliveryPhoneNumber,
    deliveryAddressSource: deliveryContact?.deliveryAddressSource,
    prescriptionHeaderIds: [prescriptionHeaderId],
  };
}
